pkg/controllers/updaterun/initialization_integration_test.go

```diff
@@ -0,0 +1,740 @@
+			Expect(k8sClient.Create(ctx, policySnapshot)).To(Succeed())
+			Expect(k8sClient.Create(ctx, snapshot2)).To(Succeed())
+
+			By("Creating a new clusterStagedUpdateRun")
+			Expect(k8sClient.Create(ctx, updateRun)).To(Succeed())
+
+			By("Validating the initialization failed")
+			validateFailedInitCondition(ctx, updateRun, "more than one (2 in actual) latest policy snapshots associated")
+
+			By("Deleting the second scheduling policy snapshot")
+			Expect(k8sClient.Delete(ctx, snapshot2)).Should(Succeed())
+		})
+
+		It("Should fail to initialize if the latest policy snapshot has a nil policy", func() {
+			By("Creating scheduling policy snapshot with nil policy")
+			policySnapshot.Spec.Policy = nil
+			Expect(k8sClient.Create(ctx, policySnapshot)).To(Succeed())
+
+			By("Creating a new clusterStagedUpdateRun")
+			Expect(k8sClient.Create(ctx, updateRun)).To(Succeed())
+
+			By("Validating the initialization failed")
+			validateFailedInitCondition(ctx, updateRun, "does not have a policy")
+		})
+
+		It("Should fail to initialize if the latest policy snapshot does not have valid cluster count annotation", func() {
+			By("Creating scheduling policy snapshot with invalid cluster count annotation")
+			delete(policySnapshot.Annotations, placementv1beta1.NumberOfClustersAnnotation)
+			Expect(k8sClient.Create(ctx, policySnapshot)).To(Succeed())
+
+			By("Creating a new clusterStagedUpdateRun")
+			Expect(k8sClient.Create(ctx, updateRun)).To(Succeed())
+
+			By("Validating the initialization failed")
+			validateFailedInitCondition(ctx, updateRun, "doesn't have valid cluster count annotation")
+		})
+
+		It("Should fail to initialize if the latest policy snapshot is not fully scheduled yet", func() {
+			By("Creating scheduling policy snapshot without scheduled condition")
+			Expect(k8sClient.Create(ctx, policySnapshot)).To(Succeed())
+
+			By("Creating a new clusterStagedUpdateRun")
+			Expect(k8sClient.Create(ctx, updateRun)).To(Succeed())
+
+			By("Validating the initialization failed")
+			validateFailedInitCondition(ctx, updateRun, "not fully scheduled yet")
+		})
+
+		It("Should copy the latest policy snapshot details to the updateRun status -- pickN policy", func() {
+			By("Creating scheduling policy snapshot with pickN policy")
+			Expect(k8sClient.Create(ctx, policySnapshot)).To(Succeed())
+
+			By("Set the latest policy snapshot condition as fully scheduled")
+			meta.SetStatusCondition(&policySnapshot.Status.Conditions, metav1.Condition{
+				Type:               string(placementv1beta1.PolicySnapshotScheduled),
+				Status:             metav1.ConditionTrue,
+				ObservedGeneration: policySnapshot.Generation,
+				Reason:             "scheduled",
+			})
+			Expect(k8sClient.Status().Update(ctx, policySnapshot)).Should(Succeed(), "failed to update the policy snapshot condition")
+
+			By("Creating a new clusterStagedUpdateRun")
+			Expect(k8sClient.Create(ctx, updateRun)).To(Succeed())
+
+			By("Validating the initialization failed")
```

Comment:
I don't seem to see where this fail? There is no validateFailedInitCondition function all.  Maybe this test case is just to verify the policySnapshot properties are copied to updateRun?

Reply:
good catch. Updated the message.